src/main/java/org/cristalise/lookup/ldap/LDAPLookup.java

```diff
@@ -435,13 +435,16 @@ public ItemPath getItemPath(String uuid) throws ObjectNotFoundException, Invalid
         try {
         	ItemPath item = new ItemPath(uuid);
             LDAPEntry anEntry=mLDAPAuth.getAuthObject().read(getDN(item)+mLocalPath,attr);
-            String type = LDAPLookupUtils.getFirstAttributeValue(anEntry, "objectClass");
-            if (type.equals("cristalentity"))
-                return item;
-            else if (type.equals("cristalagent"))
-                return new AgentPath(item);
-            else
-                throw new ObjectNotFoundException("Not an entity '"+uuid+"'");
+            String[] values = LDAPLookupUtils.getAllAttributeValues(anEntry, "objectclass");
+            if (values != null) {
+            	for (String type : values) {
+            		if (type.equals("cristalentity"))
+                        return item;
```

Comment:
I'd recommend only explicity returning if "cristalagent" is found. Otherwise for "cristalentity" set an isItem boolean and finish the iteration. Then return the item or throw not found based on the boolean.